File: crates/klyron_test/bindings/cpp/builder.cpp
#include "builder.hpp"

namespace klyron_test {
TestBuilder::TestBuilder() : config_() {}
TestBuilder& TestBuilder::enabled(bool v) { config_.enabled = v; return *this; }
TestBuilder& TestBuilder::verbose(bool v) { config_.verbose = v; return *this; }
TestConfig TestBuilder::build() { return config_; }
}

File: crates/klyron_test/bindings/rust/builder.rs
use crate::config::TestConfig;

pub struct TestBuilder {
    config: TestConfig,
}

impl TestBuilder {
    pub fn new() -> Self { Self { config: TestConfig::default() } }
    pub fn build(self) -> TestConfig { self.config }
}

impl Default for TestBuilder {
    fn default() -> Self { Self::new() }
}
